ui/app/components/job-status/deployment-history.js

```diff
@@ -0,0 +1,34 @@
+// @ts-check
+import Component from '@glimmer/component';
+import { watchRelationship } from 'nomad-ui/utils/properties/watch';
+import { inject as service } from '@ember/service';
+
+export default class JobStatusDeploymentHistoryComponent extends Component {
+  @service store;
+
+  async watchDeploymentHistory() {
+    const deployment = await this.args.deployment;
+    this.watch.perform(deployment, 10000);
+  }
+
+  willDestroy() {
+    super.willDestroy(...arguments);
+    this.watch.cancelAll();
+  }
```

Comment:
This is pretty atypical for Nomad: calling a `watcher` from anyplace other than a route. After having a good discussion with @DingoEatingFuzz , this seems like the best place to fetch this data, however - as it is both conditional and quite nested relative to the parent route.